### PR TITLE
generator: blindly rename Repository to Exercise

### PR DESCRIPTION
**Pull request by hilary**
first step in refactoring the Repository model. Renaming it to Exercise. part of exercism/xruby#633

### CODE REVIEW

File: lib/generator/command_line/generator_optparser.rb [hunk 1]
```diff
@@ -54,16 +54,16 @@ def available_generators
 
     def validate_paths
       return true if File.directory?(@paths.metadata)
-      $stderr.puts metadata_repository_missing_message(@paths.metadata)
+      $stderr.puts metadata_exercise_missing_message(@paths.metadata)
```
Comment (Insti) [inline lib/generator/command_line/generator_optparser.rb]: git_repository

File: lib/generator/files.rb [hunk 2]
```diff
@@ -7,10 +7,10 @@ def self.read(filename)
     end
 
     class Readable
-      attr_reader :filename, :repository_root
-      def initialize(filename:, repository_root: nil)
+      attr_reader :filename, :exercise_root
```
Comment (Insti) [inline lib/generator/files.rb]: should be git_repository.

File: lib/generator/files.rb [hunk 3]
```diff
@@ -47,7 +47,7 @@ def write(content)
       end
     end
 
-    # An Exercise is used as part of a Repository
+    # An Exercise is used as part of a Exercise
```
Comment (Insti) [inline lib/generator/files.rb]: Oh dear.

**Comment by hilary** [comment]
right. bad idea.

**Pull request closed without merging**